src/allocators/lib/AllocatorLib.sol

```diff
@@ -0,0 +1,218 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.27;
+
+import {ERC6909} from '@solady/tokens/ERC6909.sol';
+
+import {ITheCompact} from '@uniswap/the-compact/interfaces/ITheCompact.sol';
+import {LOCK_TYPEHASH, Lock} from '@uniswap/the-compact/types/EIP712Types.sol';
+
+library AllocatorLib {
+    // bytes4(keccak256('prepareAllocation(address,uint256[2][],address,uint256,bytes32,bytes32,bytes)'));
+    bytes4 public constant PREPARE_ALLOCATION_SELECTOR = 0x7ef6597a;
+
+    error InvalidBalanceChange(uint256 newBalance, uint256 oldBalance);
+    error InvalidPreparation();
+    error InvalidRegistration(address recipient, bytes32 claimHash, bytes32 typehash);
+
+    function prepareAllocation(
+        address compactContract,
+        uint256 nonce,
+        address recipient,
+        uint256[2][] calldata idsAndAmounts,
```

Comment:
one thing that's a little funky is that there doesn't seem to be a way to prepare an allocation for an unknown amount (e.g. fee on transfer balance post-deposit). That said, I think it's still workable in most cases through other means before you get to the preparation stage